src/libs/syncpoint/syncpoint.cpp

```diff
@@ -599,15 +601,21 @@ SyncPoint::handle_default(string component, WakeupType type)
 	                  get_identifier().c_str(),
 	                  max_waittime_sec_ + static_cast<float>(max_waittime_nsec_) / 1000000000.f);
 	bad_components_.insert(pending_emitters_.begin(), pending_emitters_.end());
-	if (bad_components_.size() > 1) {
-		string bad_components_string = "";
+	if (!bad_components_.empty()) {
+		stringstream bad_components_string;
```

Comment:
I would change the variable name to something like
```suggestion
		stringstream bad_components_msg;
```
so that it does not suggests to be a type that it is not.

Reply:
Thanks for the suggestion! I simplified the name to `message`.